dbms/src/Storages/Page/V3/Universal/UniversalWriteBatchAdaptor.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <IO/ReadBuffer.h>
+#include <IO/WriteHelpers.h>
+#include <Storages/Page/PageDefinesBase.h>
+#include <Storages/Page/V3/Universal/UniversalPageIdFormat.h>
+#include <Storages/Page/V3/Universal/UniversalWriteBatch.h>
+#include <Storages/Page/WriteBatch.h>
+
+#include <vector>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int LOGICAL_ERROR;
+} // namespace ErrorCodes
+
+// Adapt UniversalWriteBatch to WriteBatch interface
+class UniversalWriteBatchAdaptor : private boost::noncopyable
```

Comment:
I think this class can be merged into `UniversalWriteBatch`

Reply:
Merged.